test/rule_test_support.dart

```diff
@@ -388,7 +388,7 @@ abstract class _ContextResolutionTest with ResourceProviderMixin {
       throw StateError('Only dart files can be changed after analysis.');
     }
 
-    return super.newFile(path, content: content);
+    return super.newFile2(path, content);
```

Comment:
I would recommend to rename the containing method to `newFile2`, with corresponding making `content` a required positional parameter.

Reply:
Done!